factoryWithTypeCheckers.js

```diff
@@ -269,6 +288,107 @@ module.exports = function(isValidElement, throwOnDirectAccess) {
     }
     return createChainableTypeChecker(validate);
   }
+  
+  function createTupleOfTypeChecker(typeCheckerTuple) {
+    var argsInvalid = !Array.isArray(typeCheckerTuple) || !typeCheckerTuple.every(function (checker) { return typeof checker === 'function'});
+
+    function validate(props, propName, componentName, location, propFullName) {
+      if (argsInvalid) {
+        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside tupleOf.')
+      }
+      var propValue = props[propName];
+      if (!Array.isArray(propValue)) {
+        var propType = getPropType(propValue);
+        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of type ' + '`' + propType + '` supplied to `' + componentName + '`, expected an array.');
+      }
+      if (propValue.length !== typeCheckerTuple.length) {
+        return new PropTypeError('Invalid ' + location + ' `' + propFullName + '` of length ' + '`' + propValue.length + '` supplied to `' + componentName + '`, expected an array of length `' + typeCheckerTuple.length + '`.');
+      }
+      for (var i = 0; i < typeCheckerTuple.length; i++) {
+        var typeChecker = typeCheckerTuple[i];
+        var error = typeChecker(propValue, i, componentName, location, propFullName + '[' + i + ']', ReactPropTypesSecret);
+        if (error instanceof Error) {
+          return error;
+        }
+      }
+      return null;
+    }
+    return createChainableTypeChecker(validate);
+  }
+
+  function createIterableOfTypeChecker(typeChecker) {
+    function validate(props, propName, componentName, location, propFullName) {
+      if (typeof typeChecker !== 'function') {
+        return new PropTypeError('Property `' + propFullName + '` of component `' + componentName + '` has invalid PropType notation inside iterableOf.');
+      }
+      if (!(Symbol && Symbol.iterator)) {
+        return new PropTypeError('Cannot use `PropTypes.iterableOf` if Symbol.iterator is not present');
```

Comment:
technically this isn't true; you could use `Array.from` exclusively, which can be polyfilled.